include/seqan3/alignment/pairwise/alignment_algorithm.hpp

```diff
@@ -142,6 +151,115 @@ class alignment_algorithm :
         return align_result{res};
     }
 
+    /*!\brief Invokes the banded alignment computation given two sequences.
+     * \tparam    first_batch_t  The type of the first sequence (or packed sequences); must model std::ForwardRange.
+     * \tparam    second_batch_t The type of the second sequence (or packed sequences); must model std::ForwardRange.
+     * \param[in] first_batch    The first sequence (or packed sequences).
+     * \param[in] second_batch   The second sequence (or packed sequences).
+     *
+     * \details
+     *
+     * Computes the k-banded alignment. This function is only available if the alignment configuration was configured
+     * with seqan3::align_cfg::band.
+     *
+     * ### Exception
+     *
+     * Strong exception guarantee. Might throw std::bad_alloc or seqan3::invalid_alignment_configuration.
+     *
+     * ### Thread-safety
+     *
+     * Calls to this functions in a concurrent environment are not thread safe. Instead use a copy of the alignment
+     * algorithm type.
+     *
+     * ### Complexity
+     *
+     * The code always runs in \f$ O(N*k) \f$ time and depending on the configuration requires at least \f$ O(k) \f$
+     * and at most \f$ O(N*k) \f$ space.
+     */
+    template <std::ranges::ForwardRange first_batch_t, std::ranges::ForwardRange second_batch_t>
+    auto operator()(first_batch_t const & first_batch, second_batch_t const & second_batch)
+        requires is_banded
+    {
+        assert(cfg_ptr != nullptr);
+
+        using std::get;
+
+        static_assert(config_t::template exists<align_cfg::band>(),
+                      "The band configuration is required for the banded alignment algorithm.");
+        auto const & band = get<align_cfg::band>(*cfg_ptr).value;
+
+        // ----------------------------------------------------------------------------
+        // Check valid band settings.
+        // ----------------------------------------------------------------------------
+
+        using diff_type = typename std::iterator_traits<std::ranges::iterator_t<first_batch_t>>::difference_type;
+        static_assert(std::is_signed_v<diff_type>,  "Only signed types can be used to test the band parameters.");
+
+        if (static_cast<diff_type>(band.lower_bound) >
+            std::distance(seqan3::begin(first_batch), seqan3::end(first_batch)))
+        {
+            throw invalid_alignment_configuration
+            {
+                "The lower bound is bigger than the size of the first sequence."
+            };
+        }
+
+        if (static_cast<diff_type>(std::abs(band.upper_bound)) >
+            std::distance(seqan3::begin(second_batch), seqan3::end(second_batch)))
```

Comment:
Why do you use `std::abs` here now, shouldn't a negative value be kept as negative?

Reply:
Because it now tests, whether the upper bound starts behind the end of the vertical sequence. This means the value is negative and needs to be compared with the distance type between begin and end.

```
 01234567
0
1
2
3
4
5
.
.x
. x
.  x
.x  x
. x   x
```

Here the band is set to lower_bound:-10 upper_bound: -7; Hence I use the abs to get the positive value.

Reply:
Ok, I get the reason you need to change the sign, but should it then not be just `-band.upper_bound` and not abs?

Reply:
They mean and do literally the same (I think even from the issued instructions) but the abs version is more expressive in what it does, I think.